addons/purchase_stock/static/src/product_catalog/search/search_panel.js

```diff
@@ -33,14 +34,15 @@ export class PurchaseSuggestCatalogSearchPanel extends AccountProductCatalogSear
     }
     onDaysInput(ev) {
         const value = parseInt(ev.target.value, 10) || 0;
-        const bounded_val = value > 0 ? (value < 999 ? value : 999) : 0; // 999 because input is 3 digits wide
+        const bounded_val = clamp(value, 0, 999); // 999 because input is 3 digits wide
         this.suggest.numberOfDays = bounded_val;
         ev.target.value = bounded_val;
     }
     onPercentFactorInput(ev) {
         const value = parseInt(ev.target.value, 10) || 0;
-        this.suggest.percentFactor = value > 0 ? value : 0; // Negative percent doesn't make sense
-        ev.target.value = value > 0 ? value : 0;
+        const bounded_val = clamp(value, 0, 999); // 999 because input is 3 digits wide
+        this.suggest.percentFactor = bounded_val;
+        ev.target.value = bounded_val;
```

Comment:
```suggestion
        const boundedVal = clamp(value, 0, 999); // 999 because input is 3 digits wide
        this.suggest.percentFactor = boundedVal;
        ev.target.value = boundedVal;
```